### ISSUE TITLE
[Question] writing json document using simdjson?

### ISSUE THREAD

**Issue by cppchedy** [simdjson/simdjson#665]
Hi @lemire,
simdjson v0.3 is a beast thanks for your great work(and also for all contributors).

I have a questions on the ability to write json document(starting from an empty doc/element). As of now (v0.3) I don't see support  for this. so my questions is: 

- is there any plans on supporting this? 
- how much effort(time) does it  take? (asking to evaluate a possible PR)

the current work around is to use simdjson + another lib, so it would be nice if I could only rely on simdjson?

thanks

**Comment by lemire** [comment]
Correct, it is not currently possible to write (serialize) with simdjson.

@jkeiser : can you take this question at your convenience?

**Comment by jkeiser** [comment]
A write API is an important feature for the reasons you mention, and it's on our minds--but we have some things we want to finish on the read side first (streaming and SAX particularly). So probably not soon. I expect another couple of releases to pass before we really jump into the problem.

We do welcome contributions, though! If you are interested, we can always talk it out, all of us here try to be helpful :) I imagine an initial stab isn't hard if it's designed right, but it depends entirely on your experience and engagement. Probably just a "document builder" API that builds a document in the current tape format. Then people can just use the current cout << document to write it outI think it could be done without SIMD, and with a single implementation.

**Comment by eranif** [comment]
Shame it is not supported. I am very impressed with the speed of reading, but I am afraid that only having the "Read API" is not sufficient for most C++ programs

**Comment by saidwho12** [comment]
I would love to have also a write API for this, otherwise for my use case I'll probably just have to use something like RapidJSON

**Comment by slidertom** [comment]
I'm also have same issue and thinking about something like: edit mode -> RapidJSON and read-only mode simdjson.
simdjson looks like already contains some write operations like: simdjson::to_json_string for array.

**Comment by lemire** [comment]
We are openly calling for interested contributors. We don’t even have design documents at this point: what we want to do and how we are going to do it efficiently.

**Comment by SpeedyCraftah** [comment]
Would be nice if there was some kind of central discussion thread for this but this will have to do.
For now the best approach would probably be to use something like RapidJSON alongside simdjson (rapid for serializing, simdjson for reading), but of course it would be great if simdjson supported it.

I would love to help but more or less fear a lack of expertise in this type of area, but for a design something like this should likely at least be on par with RapidJSON in terms of performance, although I'm not sure how SIMD could be leveraged to accelerate this process, although that's probably the least of our problems right now.

It would probably also have to be separate from the core principles of the ondemand parser which is taking an iterative approach while an object builder would likely hold and have everything available immediately.
